sample-service/src/main/java/uk/ac/ox/ndph/mts/sample_service/security/authorisation/AuthorisationService.java

```diff
@@ -0,0 +1,109 @@
+package uk.ac.ox.ndph.mts.sample_service.security.authorisation;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.ac.ox.ndph.mts.sample_service.client.practitioner_service.PractitionerServiceClient;
+import uk.ac.ox.ndph.mts.sample_service.client.role_service.RoleServiceClient;
+import uk.ac.ox.ndph.mts.sample_service.client.dtos.RoleAssignmentDTO;
+import uk.ac.ox.ndph.mts.sample_service.client.dtos.RoleDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.List;
+
+/**
+ * The service which performs the authorisation flow
+ */
+@Service
+public class AuthorisationService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuthorisationService.class);
+
+    private final SecurityContextUtil securityContextUtil;
+
+    private final PractitionerServiceClient practitionerServiceClient;
+    private final RoleServiceClient roleServiceClient;
+
+    @Autowired
+    public AuthorisationService(final SecurityContextUtil securityContextUtil,
+                                final PractitionerServiceClient practitionerServiceClient,
+                                final RoleServiceClient roleServiceClient) {
+        this.securityContextUtil = securityContextUtil;
+        this.practitionerServiceClient = practitionerServiceClient;
+        this.roleServiceClient = roleServiceClient;
+    }
+
+    /**
+     * Authorise request
+     * @return true if authorised
+     */
+    public boolean authorise(String requiredPermission)  {
```

Comment:
Long method - 40 lines.  Could perhaps refactor into two helper methods userHasRoleWithPermission and userHasRoleAtSite ?